v0/destinations/snapchat_conversion/transform.js

```diff
@@ -0,0 +1,395 @@
+const get = require("get-value");
+const { logger } = require("handlebars");
+const sha256 = require("sha256");
+const { EventType } = require("../../../constants");
+
+const {
+  CustomError,
+  defaultPostRequestConfig,
+  constructPayload,
+  defaultRequestConfig,
+  removeUndefinedAndNullValues,
+  getFieldValueFromMessage,
+  defaultBatchRequestConfig,
+  getErrorRespEvents,
+  getSuccessRespEvents
+} = require("../../util");
+const ErrorBuilder = require("../../util/error");
+const {
+  ENDPOINT,
+  eventNameMapping,
+  mappingConfig,
+  ConfigCategory,
+  MAX_BATCH_SIZE
+} = require("./config");
+const {
+  msUnixTimestamp,
+  getItemIds,
+  getPriceSum,
+  getDataUseValue,
+  getNormalizedPhoneNumber
+} = require("./util");
+
+/**
+ * Get access token to be bound to the event req headers
+ *
+ * Note:
+ * This method needs to be implemented particular to the destination
+ * As the schema that we'd get in `metadata.secret` can be different
+ * for different destinations
+ *
+ * @param {Object} metadata
+ * @returns
+ */
+const getAccessToken = metadata => {
+  // OAuth for this destination
+  const { secret } = metadata;
+  if (!secret) {
+    throw new ErrorBuilder()
+      .setMessage("Empty/Invalid access token")
+      .setStatus(500)
+      .build();
+  }
+  return secret.access_token;
+};
+
+function trackResponseBuilder(message, metadata, { Config }) {
+  let event = get(message, "event");
+  if (!event) {
+    throw new CustomError("Event name is required", 400);
+  }
+  event = event.trim().replace(/\s+/g, "_");
+
+  const { pixelId, snapAppId, appId } = Config;
+  const eventConversionType = message?.context.snapchat_conversion_type?.toUpperCase();
```

Comment:
```suggestion
  const eventConversionType = message?.context?.snapchat_conversion_type?.toUpperCase();
```

Reply:
Is it the right place to take snapchat_conversion_type? CC @utsabc

Reply:
It is decided we will use `OFFLINE` by default. Before that, we will check in `properties.eventConversionType`, then `channel`.